### PR TITLE
[RDY] Small town buildings

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
Added small town buildings from http://smf.cataclysmdda.com/index.php?topic=14434.0.

Thanks to Whaley (http://smf.cataclysmdda.com/index.php?action=profile;u=3470).

### Shops

- `bakery` (2 variants);
- `candy_shop` (2 variants);
- `construction_site` (6 variants);
- `icecream_shop`;
- `post_office` (2 variants).

### Overmap specials

- `lumber_yard` (4 tiles).

***

I've decided to move `pet_store` and `zoo` to separate PR.

### CODE REVIEW

**Comment by John-Candlebury** [comment]
Animals in the zoo should probably be zombies or dead

**Comment by DeadLeaves** [comment]
How's this comming along? I'm really looking forward to this PR.

File: data/json/mapgen/small_town_buildings/construction_site.json [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+        "T": "f_table",
+        "b": "f_barricade_road",
+        "c": "f_chair",
+        "{": "f_dumpster"
+      },
+      "place_items": [
+        { "item": "tools_carpentry", "x": 15, "y": 11, "chance": 30 },
+        { "item": "power_tools", "x": 9, "y": 12, "chance": 10 },
+        { "item": "power_tools", "x": 8, "y": 9, "chance": 10 },
+        { "item": "clothing_work_hat", "x": 15, "y": 8, "chance": 0 },
```
Comment (Coolthulhu) [inline data/json/mapgen/small_town_buildings/construction_site.json]: 0 chance?

File: data/json/mapgen/small_town_buildings/lumberyard_1_1.json [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        { "item": "2x4", "x": 12, "y": 17 },
+        { "item": "2x4", "x": 14, "y": 17 },
+        { "item": "2x4", "x": 5, "y": 19 },
+        { "item": "2x4", "x": 6, "y": 19 },
+        { "item": "2x4", "x": 7, "y": 19 },
+        { "item": "2x4", "x": 9, "y": 19 },
+        { "item": "2x4", "x": 12, "y": 19 },
+        { "item": "2x4", "x": 13, "y": 19 },
+        { "item": "2x4", "x": 14, "y": 19 },
+        { "item": "2x4", "x": 15, "y": 19 }
```
Comment (Coolthulhu) [inline data/json/mapgen/small_town_buildings/lumberyard_1_1.json]: A ton of hard-set planks will look bad. Those should be spawned over an area with rolled x, y and repeat.

File: data/json/mapgen/small_town_buildings/pet_store.json [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        { "item": "pet_carrier", "x": 8, "y": 14 }
+      ],
+      "place_monster": [
+        { "monster": "mon_cat", "x": 2, "y": 3 },
+        { "monster": "mon_rabbit", "x": 2, "y": 7 },
+        { "monster": "mon_rabbit", "x": 3, "y": 7 },
+        { "monster": "mon_dog_zombie_rot", "x": 2, "y": 9 },
+        { "monster": "mon_dog", "x": 2, "y": 13 },
+        { "monster": "mon_dog", "x": 2, "y": 15 },
+        { "monster": "mon_dog_zombie_rot", "x": 10, "y": 19 }
```
Comment (Coolthulhu) [inline data/json/mapgen/small_town_buildings/pet_store.json]: The animals should be rolled from a group, not predefined.

File: data/json/mapgen/small_town_buildings/pet_store.json [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        { "item": "catfood", "x": 8, "y": 7 },
+        { "item": "catfood", "x": 8, "y": 8 },
+        { "item": "pet_carrier", "x": 21, "y": 9 },
+        { "item": "dogfood", "x": 8, "y": 10 },
+        { "item": "pet_carrier", "x": 21, "y": 10 },
+        { "item": "dogfood", "x": 8, "y": 11 },
+        { "item": "dogfood", "x": 15, "y": 12 },
+        { "item": "catfood", "x": 16, "y": 12 },
+        { "item": "catfood", "x": 17, "y": 12 },
+        { "item": "pet_carrier", "x": 8, "y": 14 }
```
Comment (Coolthulhu) [inline data/json/mapgen/small_town_buildings/pet_store.json]: The items should be a group, unless there is a good reason to place exact types.

File: data/json/mapgen/small_town_buildings/zoo_0_2.json [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        { "item": "stick", "x": 7, "y": 11 },
+        { "item": "rock", "x": 18, "y": 13 },
+        { "item": "pine_bough", "x": 7, "y": 14 },
+        { "item": "stick", "x": 9, "y": 14 },
+        { "item": "stick", "x": 18, "y": 15 },
+        { "item": "rock", "x": 12, "y": 16 }
+      ],
+      "place_monster": [
+        { "monster": "mon_bobcat", "x": 8, "y": 5 },
+        { "monster": "mon_spider_jumping", "x": 18, "y": 9 },
```
Comment (Coolthulhu) [inline data/json/mapgen/small_town_buildings/zoo_0_2.json]: Giant jumping spider in normal zoo?

File: data/json/mapgen/small_town_buildings/zoo_2_1.json [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        "b": "f_bench",
+        "c": "f_counter",
+        "f": "f_dahlia"
+      },
+      "toilets": {
+        "&": {}
+      },
+      "place_monsters": [
+        { "monster": "GROUP_FOREST", "x": 15, "y": 19, "chance": 6, "density": 0 },
+        { "monster": "GROUP_FOREST", "x": 4, "y": 19, "chance": 6, "density": 0 }
```
Comment (Coolthulhu) [inline data/json/mapgen/small_town_buildings/zoo_2_1.json]: Density 0?

File: data/json/overmap_terrain.json [hunk 7]
```diff
@@ -7710,5 +7710,392 @@
...
+    "mondensity": 2
+  },
+  {
+    "id": "state_park_0_1",
+    "type": "overmap_terrain",
+    "name": "state park",
+    "sym": 83,
+    "color": "i_green",
+    "see_cost": 5,
+    "mondensity": 2
```
Comment (Coolthulhu) [inline data/json/overmap_terrain.json]: Should use `copy-from` instead of redefining things.

**Comment by ZhilkinSerg** [comment]
> How's this comming along? I'm really looking forward to this PR.

Slowly 🐢 

default

**Comment by nornagon** [comment]
Ah, it looks like we're both adding a bike shop: #22262. We should merge them into variants :)

**Comment by ZhilkinSerg** [comment]
> Ah, it looks like we're both adding a bike shop: #22262. We should merge them into variants :)

I will bike shop to separate PR. Thanks.

File: data/json/mapgen/small_town_buildings/lumberyard_1_1.json (hunk 2 above)
Comment (ZhilkinSerg) [reply data/json/mapgen/small_town_buildings/lumberyard_1_1.json]: Fixed this.

File: data/json/mapgen/small_town_buildings/construction_site.json (hunk 1 above)
Comment (ZhilkinSerg) [reply data/json/mapgen/small_town_buildings/construction_site.json]: Fixed this

File: data/json/overmap_terrain.json (hunk 7 above)
Comment (ZhilkinSerg) [reply data/json/overmap_terrain.json]: Fixed this

**Pull request merged**